dpctl/apis/include/syclinterface.h

```diff
@@ -0,0 +1,44 @@
+//=== syclinterace.h - single include header for libsyclinterface  -*-C-*- ===//
+//
+//                      Data Parallel Control (dpctl)
+//
+// Copyright 2020-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file includes all the headers of syclinterface/
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+// clang-format off
```

Comment:
why format off?

Reply:
Because clang would reorder includes and include C-API headers before including `dpctl_sycl_types` and then the opaque reference identifiers would be unknown and compilation fails.